compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp

```diff
@@ -1803,6 +1806,221 @@ std::optional<StringRef> LoweringConfigAttr::getLoweringStrategy() const {
+
+  for (auto [loopRange, givenTileSize] :
+       llvm::zip_equal(loopRanges, givenTileSizes)) {
+    if (isZeroInteger(givenTileSize)) {
+      offsets.push_back(loopRange.offset);
+      sizes.push_back(loopRange.size);
+    } else {
+      offsets.push_back(*ivIt++);
+      sizes.push_back(givenTileSize);
+      ids.push_back(sizes.size() - 1);
+    }
+  }
+  return {offsets, sizes, ids};
+}
+
+FailureOr<mlir::scf::SCFTilingOptions::CustomLoopHeaderInfo>
+ConditionalTransposeAttr::generateLoopHeaderFn(
+    OpBuilder &builder, Location loc, ArrayRef<Range> loopRanges,
+    ArrayRef<OpFoldResult> givenTileSizes,
+    ValueRange destinationTensors) const {
+  scf::ForallOp forallOp;
+  // Get loop bounds
+  SmallVector<OpFoldResult> lbs, ubs, steps;
+  std::tie(lbs, ubs, steps) = getLoopBounds(loopRanges, givenTileSizes);
+
+  if (lbs.size() != 2)
+    return emitError(loc) << "conditional_transpose only supports rank 2";
+
+  SmallVector<Attribute> deviceMappingAttribute;
+  deviceMappingAttribute.push_back(IREE::Codegen::WorkgroupMappingAttr::get(
+      builder.getContext(), IREE::Codegen::symbolizeWorkgroupId(1).value()));
+  deviceMappingAttribute.push_back(IREE::Codegen::WorkgroupMappingAttr::get(
+      builder.getContext(), IREE::Codegen::symbolizeWorkgroupId(0).value()));
+
+  std::optional<ArrayAttr> mappingAttr =
+      builder.getArrayAttr(deviceMappingAttribute);
+
+  // Compute condition for transpose
+  Value nbCusVal = builder.create<arith::ConstantIndexOp>(loc, getNbCus());
+  Value nbXCDs = builder.create<arith::ConstantIndexOp>(loc, getNbXcds());
+  Value cond = getCondition(builder, loc, ubs, steps, nbXCDs, nbCusVal);
+
+  // Apply condition on loop bounds
+  swapIf(builder, loc, cond, lbs);
+  swapIf(builder, loc, cond, ubs);
+  swapIf(builder, loc, cond, steps);
+
+  forallOp = scf::ForallOp::create(builder, loc, lbs, ubs, steps,
+                                   destinationTensors, mappingAttr);
+  builder.setInsertionPoint(forallOp.getTerminator());
+
+  SmallVector<Value> ivs = forallOp.getInductionVars();
+  SmallVector<OpFoldResult> offsets, sizes;
+  SmallVector<size_t> ids;
+  std::tie(offsets, sizes, ids) =
+      computeOffsetAndSize(loopRanges, givenTileSizes, ivs);
+
+  // Apply condition on offsets
+  swapIf(builder, loc, cond, offsets, ids);
+
+  ValueRange innerDestinationTensors = forallOp.getRegionOutArgs();
+  return mlir::scf::SCFTilingOptions::CustomLoopHeaderInfo{
+      {cast<LoopLikeOpInterface>(forallOp.getOperation())},
+      offsets,
+      sizes,
+      innerDestinationTensors};
+}
+
+LogicalResult ConditionalTransposeAttr::generateLoopTerminatorFn(
+    OpBuilder &builder, Location loc, ArrayRef<LoopLikeOpInterface> loops,
+    ValueRange tiledResults, ArrayRef<SmallVector<OpFoldResult>> resultOffsets,
+    ArrayRef<SmallVector<OpFoldResult>> resultSizes,
+    ValueRange destinationTensors) const {
+
+  assert(loops.size() == 1 && "Expected loop count should be 1.");
```

Comment:
```suggestion
    ValueRange destinationTensors) const {
  assert(loops.size() == 1 && "Expected loop count should be 1.");
```
nit: superfluous empty line